tools/astarte_export/lib/fetchdata/fetchdata.ex

```diff
@@ -0,0 +1,333 @@
+          reception_timestamp =
+            map[:reception_timestamp]
+            |> DateTime.from_unix!(:millisecond)
+            |> DateTime.to_iso8601()
+
+          list = Map.to_list(map)
+
+          value_list =
+            List.foldl(list, [], fn {key, value}, acc1 ->
+              with "v_" <> item <- to_string(key),
+                   match_object when match_object != nil <-
+                     Enum.find(extract_2nd_level_params, fn map1 -> map1[:suffix_path] == item end),
+                   data_type = match_object[:data_type],
+                   token = "/" <> match_object[:suffix_path],
+                   value1 when value1 != "" <- from_native_type(value, data_type) do
+                [%{attributes: [name: token], value: value1} | acc1]
+              else
+                _ -> acc1
+              end
+            end)
+
+          acc ++ [%{attributes: [reception_timestamp: reception_timestamp], value: value_list}]
+        end)
+
+      {:more_data, values, updated_options}
+    else
+      {:error, reason} -> {:error, reason}
+    end
+  end
+
+  def fetch_individual_properties(conn, realm, mapping, device_id, interface_id, options) do
+    endpoint_id = mapping.endpoint_id
+    path = mapping.endpoint
+    data_type = mapping.value_type
+    data_field = CQLUtils.type_to_db_column_name(data_type)
+
+    with {:ok, result} <-
+           Queries.retrieve_individual_properties(
+             conn,
+             realm,
+             device_id,
+             interface_id,
+             data_field,
+             options
+           ),
+         [value | _] = result_list <- Enum.to_list(result) do
+      updated_options = Keyword.put(options, :paging_state, result.paging_state)
+
+      values =
+        Enum.map(result_list, fn map ->
+          reception_timestamp =
+            map.reception_timestamp
+            |> DateTime.from_unix!(:millisecond)
+            |> DateTime.to_iso8601()
+
+          path = map.path |> Kernel.to_string()
+
+          atom_data_field = String.to_atom(data_field)
+          return_value = map[atom_data_field]
+          value = from_native_type(return_value, data_type)
+
+          %{attributes: [reception_timestamp: reception_timestamp, path: path], value: value}
+        end)
+
+      {:more_data, values, updated_options}
+    else
+      [] -> {:ok, :completed}
+      {:error, reason} -> {:error, reason}
+    end
+  end
+
+  defp from_native_type(value_chars, :double) do
+    to_string(value_chars)
+  end
+
+  defp from_native_type(value_chars, :integer) do
+    to_string(value_chars)
+  end
+
+  defp from_native_type(value_chars, :boolean) do
+    to_string(value_chars)
+  end
+
+  defp from_native_type(value_chars, :longinteger) do
+    to_string(value_chars)
+  end
+
+  defp from_native_type(value_chars, :string) do
+    to_string(value_chars)
+  end
+
+  defp from_native_type(value_chars, :binaryblob) do
+    base64 = to_string(value_chars)
```

Comment:
Here `to_string` is just useless. If you need to base64 encode it, it means that it is already a binary.